devtron-labs/devtron
# feat: Gitops or helm options for cd

# Description

Added GitOps or Helm deployment option for CD. Earlier if GitOps was configured then apps were deployed using GitOps by default, but now user will have option to choose between GitOps or Helm. GitOps and Helm option can be disabled by using environment variable.

Fixes # (issue)

## Type of change


# How Has This Been Tested?

For testing :- 

1) Check if GitOps/Helm option is available when creating CD pipeline. Also check that this cannot be changed during update.
2) Check if GitOps/Helm option is available when deploying Helm app. Check that this cannot be changed once Helm app is deployed i.e during update.

Unit test cases are also written for testing business logic.


# Checklist:
